webapp/channels/src/components/profile_popover_controller/index.tsx

```diff
@@ -0,0 +1,161 @@
+
+    /**
+     * Function to call to return focus to the previously focused element when the popover closes.
+     * If not provided, the popover will automatically determine the previously focused element
+     * and focus that on close. However, if the previously focused element is not correctly detected
+     * by the popover, or the previously focused element will disappear after the popover opens,
+     * it is necessary to provide this function to focus the correct element.
+     */
+    returnFocus?: () => void;
+}
+
+function ProfilePopoverController<TriggerComponentType = HTMLSpanElement>(props: Props<TriggerComponentType>) {
+    const [isOpen, setOpen] = useState(false);
+
+    const {refs, floatingStyles, context: floatingContext} = useFloating({
+        open: isOpen,
+        onOpenChange: setOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [autoPlacement()],
+    });
+
+    const {isMounted, styles: transitionStyles} = useTransitionStyles(floatingContext, {
+        duration: {
+            open: PROFILE_POPOVER_OPENING_DELAY,
+            close: PROFILE_POPOVER_CLOSING_DELAY,
+        },
+    });
+
+    const combinedFloatingStyles = Object.assign({}, floatingStyles, transitionStyles);
+
+    const clickInteractions = useClick(floatingContext);
+
+    const dismissInteraction = useDismiss(floatingContext);
+
+    const role = useRole(floatingContext);
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([
+        clickInteractions,
+        dismissInteraction,
+        role,
+    ]);
+
+    function handleHide() {
+        setOpen(false);
+    }
+
+    const TriggerComponent = props.triggerComponentRootProps?.as ?? 'span';
+
+    return (
+        <>
+            <TriggerComponent
+                id={props.triggerComponentRootProps?.id}
+                ref={refs.setReference}
+                className={props.triggerComponentRootProps?.className}
+                style={props.triggerComponentRootProps?.style}
+                {...getReferenceProps()}
+            >
+                {props.children}
+            </TriggerComponent>
+
+            {isMounted && (
+                <FloatingPortal id='user-profile-popover-portal'>
+                    <FloatingOverlay
+                        id='user-profile-popover-floating-overlay'
+                        className='user-profile-popover-floating-overlay'
+                        lockScroll={true}
+                    >
+                        <FloatingFocusManager context={floatingContext}>
+                            <div
+                                ref={refs.setFloating}
+                                style={combinedFloatingStyles}
+                                className={classNames('user-profile-popover', A11yClassNames.POPUP)}
+                                {...getFloatingProps()}
+                                id='user-profile-popover' // This is placed here as id is being overwritten by getFloatingProps
```

Comment:
No it does not need to be unique because at a time only one profile popover will be opened, reason i added this is because it was having id as `floating-ui-something` i thought this was a better one instead of that